libraries/AP_Compass/AP_Compass.h

```diff
@@ -396,5 +387,17 @@ friend class AP_Compass_Backend;
     // if we want HIL only
     bool _hil_mode:1;
 
+    // compass cal
+    bool _accept_calibration(uint8_t i);
+    bool _accept_calibration_mask(uint8_t mask);
+    void _cancel_calibration(uint8_t i);
+    void _cancel_calibration_mask(uint8_t mask);
+    uint8_t _get_cal_mask() const;
+    bool _start_calibration(uint8_t i, bool retry=false, float delay_sec=0.0f);
+    bool _start_calibration_mask(uint8_t mask, bool retry=false, bool autosave=false, float delay_sec=0.0f, bool autoreboot=false);
+
+
+    bool _auto_reboot() { return _compass_cal_autoreboot; }
```

Comment:
The idea was that all the internal interfaces that aren't exposed/used should be hidden away. The start all is the only interface used elsewhere. I could see the argument for allowing access to the mask one, however the individual compass ones shouldn't be called by any other source, there is prep work that needs to be setup, as well as coordination with the other calibrators.